onadata/apps/viewer/models/parsed_instance.py

```diff
@@ -292,6 +293,12 @@ def to_dict_for_mongo(self):
         if isinstance(self.instance.deleted_at, datetime.datetime):
             data[DELETEDAT] = self.instance.deleted_at.strftime(MONGO_STRFTIME)
 
+        if self.get_review_status():
+            data[REVIEW_STATUS] = self.get_review_status()
```

Comment:
doesn't this (and below 299-300) create a race condition? After the 296 check, someone could remove the review status before it gets to line 297. We can but this into a var before hand, also remove performance deficit of calling this twice